tests/common/portstat_utilities.py

```diff
@@ -0,0 +1,77 @@
+import logging
+
+logger = logging.getLogger('__name__')
+
+
+def parse_column_positions(separation_line, separation_char='-'):
+    '''Parse the position of each columns in the command output
+
+    Args:
+        separation_line (string): The output line separating actual data and column headers
+        separation_char (str, optional): The character used in separation line. Defaults to '-'.
+
+    Returns:
+        [list]: A list. Each item is a tuple with two elements. The first element is start position of a column. The
+                second element is the end position of the column.
+    '''
+    prev = ' ',
+    positions = []
+    for pos, char in enumerate(separation_line + ' '):
+        if char == separation_char:
+            if char != prev:
+                left = pos
+        else:
+            if char != prev:
+                right = pos
+                positions.append((left, right))
+        prev = char
+    return positions
+
+
+def parse_portstat(content_lines):
```

Comment:
Is there certain level of duplication with . https://github.com/Azure/sonic-mgmt/blob/master/tests/common/devices/sonic.py#L1256?